eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxo.hs

```diff
@@ -108,11 +108,16 @@ data BabbageUtxoPred era
   | UnequalCollateralReturn !Coin !Coin
   | DanglingWitnessDataHash !(Set.Set (DataHash (Crypto era)))
   | MalformedScripts !(Set (ScriptHash (Crypto era)))
+  | -- | list of supplied transaction outputs that are too small,
+    -- together with the minimum value for the given output.
+    BabbageOutputTooSmallUTxO
+      ![(Core.TxOut era, Integer)]
 
```

Comment:
Could we please include the Hash of the TxBody here as well.  It is often impossible to determine which Tx an predicate failure arise from. This would be very useful in may paces, but at least we have the opportunity to do it here.

Reply:
that's a bigger refactor than I was looking to do in this PR... at least if we want any kind of consistency... maybe we can make that a stretch goal in a follow up PR? this is helpful for us when testing, but is probably not actually helpful for most users, who get the predicate failure after locally submitting a single Tx.